source/fundamentals/crud/query-document.txt

```diff
@@ -0,0 +1,331 @@
+
+   .. code-block:: go
+   
+      filter := bson.D{
+	    {"$and",
+		    bson.A{
+				bson.D{{"type", "Oolong"}},
+				bson.D{{"rating", 7}},
+			}},
+      }
+
+Element
+-------
+
+Element operators check for the presence or type of the specified field.
+
+Example
+~~~~~~~
+
+The following example matches documents where the ``vendor`` field does
+not exist:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/query.go
+   :language: go
+   :dedent:
+   :start-after: begin element
+   :end-before: end element
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{_id ObjectID("...")} {type English Breakfast} {rating 6}]
+   [{_id ObjectID("...")} {type Assam} {rating 5}]
+   
+For a full list of element operators, see the :manual:`Element
+Query Operators </reference/operator/query-element/>` page.
+
+Evaluation
+----------
+
+Evaluation operators analyze values in a document based on the
+specified value in your match criteria. Common evaluation operators
+include ``$regex`` where a field's value must match the specified
+regular expression and ``$text`` where the field's value must contain
+the specified string.
+
+Example
+~~~~~~~
+
+The following example matches documents where the ``type`` begins with
+the letter "E":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/query.go
+   :language: go
+   :dedent:
+   :start-after: begin evaluation
+   :end-before: end evaluation
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{_id ObjectID("...")} {type English Breakfast} {rating 6}]
+   [{_id ObjectID("...")} {type Earl Grey} {rating 8} {vendor [A B]}]
+   
+For a full list of evaluation operators, see the :manual:`Evaluation
+Query Operators </reference/operator/query-evaluation/>` page.
+
+Array
+-----
+
+Array operators check the values or amount of elements in an array field.
+
+Example
+~~~~~~~
+
+The following example matches documents where the ``vendor`` contains "C":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/query.go
+   :language: go
+   :dedent:
+   :start-after: begin array
+   :end-before: end array
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{_id ObjectID("...")} {type Masala} {rating 10} {vendor [A C]}]
+   [{_id ObjectID("...")} {type Oolong} {rating 7} {vendor [C]}]
+
+For a full list of array operators, see the :manual:`Array
+Query Operators </reference/operator/query-array/>` page.
+
+Bitwise
+-------
+
+Bitwise operators convert a numeric field from a base 10 (decimal)
+number into its base 2 (binary) number. They check whether the value in
+a document has the same bits set as value in your match criteria.
+
+Example
+~~~~~~~
+
+The following example matches documents where the ``rating`` has the same
+bits set as ``6`` (i.e. "00000110"):
```

Comment:
Avoid Latin phrases and abbreviations, as they don't translate well and can be difficult for ESL readers.
```suggestion
bits set as ``6`` (which is "00000110"):
```